src/main/java/org/wildfly/security/auth/server/MechanismConfiguration.java

```diff
@@ -41,14 +43,19 @@
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public final class MechanismConfiguration {
+
+    private final MechanismConfiguration parent;
```

Comment:
That is all the builder actually provides, a default or a server specific definition.  The hierarchy just makes it a lot easier to use the same class for both representations and from any instance you can call forServer and it will pass up the call to the parent to select which one which will either be a server specific one if the definition exists or the default if it does not.